runtime/src/accounts_db.rs

```diff
@@ -3773,15 +3779,44 @@ impl AccountsDb {
         let scan_and_hash = || {
             let mut stats = HashStats::default();
             // When calculating hashes, it is helpful to break the pubkeys found into bins based on the pubkey value.
+            // More bins means smaller vectors to sort, copy, etc.
             const PUBKEY_BINS_FOR_CALCULATING_HASHES: usize = 64;
-            let result = Self::scan_snapshot_stores(
-                storages,
-                simple_capitalization_enabled,
-                &mut stats,
-                PUBKEY_BINS_FOR_CALCULATING_HASHES,
-            );
 
-            AccountsHash::rest_of_hash_calculation(result, &mut stats)
+            // # of passes should be a function of the total # of accounts that are active.
+            // higher passes = slower total time, lower dynamic memory usage
+            // lower passes = faster total time, higher dynamic memory usage
+            // passes=2 cuts dynamic memory usage in approximately half.
+            let passes: usize = 2;
```

Comment:
Maybe this can be passed in as a validator option so we can dynamically adjust it?

Reply:
Edit: Never mind, I see that adjusting this `passes` value will result in a different hash, so in order for this be consistent it seems we'll need to make a new release to update this value anyways.

Reply:
The hash will be the same, regardless of the # of passes we use. This should be dynamically adjustable somehow. My thinking was this isn't urgent yet, but is valuable to have ready as we scale to more accounts. Perhaps we want it calculated somehow based on the current # of accounts.

Reply:
But, it could be passed in through the validator in a subsequent pr.